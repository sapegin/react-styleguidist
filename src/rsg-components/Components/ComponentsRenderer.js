import React, { PropTypes } from 'react';

export default function ComponentsRenderer({ children }) {
	return (
		<div>
			{children}
		</div>
	);
}
ComponentsRenderer.propTypes = {
	children: PropTypes.node.isRequired,
};
