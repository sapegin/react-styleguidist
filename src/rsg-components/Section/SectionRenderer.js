import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import SectionHeading from 'rsg-components/SectionHeading';

const styles = ({ space }) => ({
	root: {
		marginBottom: space[4],
	},
});

export function SectionRenderer(allProps) {
	const { classes, name, slug, content, components, sections, primary } = allProps;
	return (
		<section className={classes.root}>
			{name &&
				<SectionHeading primary={primary} id={slug} slotName="sectionToolbar" slotProps={allProps}>
					{name}
				</SectionHeading>}
			{content}
			{components}
			{sections}
		</section>
	);
}

SectionRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	name: PropTypes.string,
	slug: PropTypes.string,
	content: PropTypes.node,
	components: PropTypes.node,
	sections: PropTypes.node,
	isolated: PropTypes.bool,
};

export default Styled(styles)(SectionRenderer);
