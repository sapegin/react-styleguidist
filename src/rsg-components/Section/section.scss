.section {

  &--standalone {
	padding: 12px;
  }

  &--flex {
	display: flex;
	flex-direction: row;
	align-items: stretch;
  }

  &__lhs {
	position: relative;
	width: 60%;
	padding: 20px 32px 0px 0px;

  }

  &__rhs {
	position: relative;
	width: 40%;
	background: #424242;
	color: whitesmoke;
	padding-top: 20px;
  }

}
