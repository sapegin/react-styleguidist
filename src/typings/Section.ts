import { Component } from './Component';
import { EXPAND_MODE } from '../scripts/schemas/config';

export interface Section {
	name?: string;
	visibleName?: string;
	components?: Component[];
	sections?: Section[];
	ignore?: string | string[];
	content?: string;
	sectionDepth?: number;
	description?: string;
	exampleMode?: EXPAND_MODE;
	usageMode?: EXPAND_MODE;
	slug?: string;
	filepath?: string;
	href?: string;
	external?: string;
}
