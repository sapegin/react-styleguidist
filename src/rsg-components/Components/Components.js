import React from 'react';
import PropTypes from 'prop-types';
import ReactComponent from 'rsg-components/ReactComponent';
import ComponentsRenderer from 'rsg-components/Components/ComponentsRenderer';

export default function Components({ components, depth, exampleMode, usageMode }) {
	return (
		<ComponentsRenderer>
			{components.map(component => (
				<ReactComponent
					key={component.filepath}
					component={component}
					exampleMode={exampleMode}
					usageMode={usageMode}
					depth={depth}
				/>
			))}
		</ComponentsRenderer>
	);
}

Components.propTypes = {
	components: PropTypes.array.isRequired,
	depth: PropTypes.number.isRequired,
	exampleMode: PropTypes.string.isRequired,
	usageMode: PropTypes.string.isRequired,
};
