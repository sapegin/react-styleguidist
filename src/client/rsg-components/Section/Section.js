import React from 'react';
import PropTypes from 'prop-types';
import Examples from 'rsg-components/Examples';
import Components from 'rsg-components/Components';
import Sections from 'rsg-components/Sections';
import SectionRenderer from 'rsg-components/Section/SectionRenderer';
import { useStyleGuideContext } from 'rsg-components/Context';
import { DisplayModes } from '../../consts';

export default function Section({ section, depth }) {
	const {
		displayMode,
		config: { pagePerSection },
	} = useStyleGuideContext();
	const {
		name,
		slug,
		filepath,
		content,
		components,
		sections,
		description,
		exampleMode,
		usageMode,
	} = section;

	const contentJsx = content && (
		<Examples examples={content} name={name} exampleMode={exampleMode} />
	);
	const componentsJsx = components && (
		<Components
			usageMode={usageMode}
			exampleMode={exampleMode}
			components={components}
			depth={depth + 1}
		/>
	);
	const sectionsJsx = sections && <Sections sections={sections} depth={depth + 1} />;

	return (
		<SectionRenderer
			description={description}
			pagePerSection={pagePerSection}
			name={name}
			slug={slug}
			filepath={filepath}
			content={contentJsx}
			components={componentsJsx}
			sections={sectionsJsx}
			isolated={displayMode !== DisplayModes.all}
			depth={depth}
		/>
	);
}

Section.propTypes = {
	section: PropTypes.object.isRequired,
	depth: PropTypes.number.isRequired,
};
