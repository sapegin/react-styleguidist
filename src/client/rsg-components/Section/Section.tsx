import React from 'react';
import PropTypes from 'prop-types';
import Examples, { ExampleModel } from 'rsg-components/Examples';
import Components from 'rsg-components/Components';
import Sections from 'rsg-components/Sections';
import SectionRenderer from 'rsg-components/Section/SectionRenderer';
import { useStyleGuideContext } from 'rsg-components/Context';
import { ComponentViewModel } from 'rsg-components/ReactComponent';
import { DisplayModes } from '../../consts';

export interface SectionViewModel {
	name?: string;
	slug?: string;
	href?: string;
	filepath?: string;
	description?: 'string';
	sections?: SectionViewModel[];
	components?: ComponentViewModel[];
	content?: ExampleModel[] | string;
	exampleMode?: string;
	usageMode?: string;
	sectionDepth?: number;
}

const Section: React.FunctionComponent<{
	section: SectionViewModel;
	depth: number;
}> = ({ section, depth }) => {
	const {
		displayMode,
		config: { pagePerSection },
	} = useStyleGuideContext();
	const {
		name,
		slug,
		filepath,
		content,
		components,
		sections,
		description,
		exampleMode,
		usageMode,
	} = section;

	const contentJsx = Array.isArray(content) ? (
		<Examples examples={content} name={name} exampleMode={exampleMode} />
	) : null;
	const componentsJsx = components && (
		<Components
			usageMode={usageMode}
			exampleMode={exampleMode}
			components={components}
			depth={depth + 1}
		/>
	);

	const sectionsJsx = sections && <Sections sections={sections} depth={depth + 1} />;

	return (
		<SectionRenderer
			description={description}
			pagePerSection={pagePerSection}
			name={name}
			slug={slug}
			filepath={filepath}
			content={contentJsx}
			components={componentsJsx}
			sections={sectionsJsx}
			isolated={displayMode !== DisplayModes.all}
			depth={depth}
		/>
	);
};

Section.propTypes = {
	section: PropTypes.any.isRequired,
	depth: PropTypes.number.isRequired,
};

export default Section;
