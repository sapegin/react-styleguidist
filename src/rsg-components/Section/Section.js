import React from 'react';
import PropTypes from 'prop-types';
import Examples from 'rsg-components/Examples';
import Components from 'rsg-components/Components';
import Sections from 'rsg-components/Sections';
import SectionRenderer from 'rsg-components/Section/SectionRenderer';

export default function Section({ section, primary }, { isolatedSection = false }) {
	const { name, slug, content, components, sections } = section;

	const contentJsx = content && <Examples examples={content} name={name} />;
	const componentsJsx = components && <Components components={components} />;
	const sectionsJsx = sections && <Sections sections={sections} />;

	return (
		<SectionRenderer
			name={name}
			slug={slug}
			content={contentJsx}
			components={componentsJsx}
			sections={sectionsJsx}
			isolated={isolatedSection}
			primary={primary}
		/>
	);
}

Section.propTypes = {
	section: PropTypes.object.isRequired,
	primary: PropTypes.bool,
};

Section.contextTypes = {
	isolatedSection: PropTypes.bool,
};
