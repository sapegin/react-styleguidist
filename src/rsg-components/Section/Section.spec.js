import test from 'ava';
import React from 'react';
import noop from 'lodash/noop';
import Examples from '../Examples';
import Components from '../Components';
import Section from './Section';
import SectionRendererHoC, { SectionRenderer } from './SectionRenderer';

const section = {
	name: 'Foo',
	content: [
		{
			type: 'code',
			content: '<button>OK</button>',
			evalInContext: noop,
		},
		{
			type: 'markdown',
			content: 'Hello *world*!',
		},
	],
	components: [],
};

test('should render component renderer', () => {
	const actual = shallow(
		<Section
			section={section}
		/>
	);

	expect(actual.node, 'to contain',
		<SectionRendererHoC
			name={section.name}
			content={<Examples examples={section.content} />}
			components={<Components components={[]} sections={[]} />}
		/>
	);
});

test('render should render component', () => {
	const actual = shallow(
		<SectionRenderer
			classes={{}}
			name={section.name}
			content={<Examples examples={section.content} />}
			components={<Components components={[]} sections={[]} />}
		/>
	);

	expect(actual.node, 'to contain',
		<h2>{section.name}</h2>
	);
	expect(actual.node, 'to contain',
		<Examples examples={section.content} />
	);
	expect(actual.node, 'to contain',
		<Components components={[]} sections={[]} />
	);
});
