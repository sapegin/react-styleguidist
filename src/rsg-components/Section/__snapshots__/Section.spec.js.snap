// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should not render title if name is not set 1`] = `<section />`;

exports[`render should render component 1`] = `
<section>
  <SectionHeading
    id="foo"
    slotName="sectionToolbar"
    slotProps={
      Object {
        "classes": Object {},
        "components": <Components
          components={Array []}
      />,
        "content": <Examples
          examples={
              Array [
                  Object {
                    "content": "<button>OK</button>",
                    "evalInContext": [Function],
                    "type": "code",
                  },
                  Object {
                    "content": "Hello *world*!",
                    "type": "markdown",
                  },
                ]
          }
          name="Foo"
      />,
        "name": "Foo",
        "sections": <Sections
          sections={Array []}
      />,
        "slug": "foo",
      }
    }
  >
    Foo
  </SectionHeading>
  <Examples
    examples={
      Array [
        Object {
          "content": "<button>OK</button>",
          "evalInContext": [Function],
          "type": "code",
        },
        Object {
          "content": "Hello *world*!",
          "type": "markdown",
        },
      ]
    }
    name="Foo"
  />
  <Components
    components={Array []}
  />
  <Sections
    sections={Array []}
  />
</section>
`;

exports[`render should render primary title if primary is set 1`] = `
<section>
  <SectionHeading
    id="test"
    primary={true}
    slotName="sectionToolbar"
    slotProps={
      Object {
        "classes": Object {},
        "name": "test",
        "primary": true,
        "slug": "test",
      }
    }
  >
    test
  </SectionHeading>
</section>
`;

exports[`render should render title if name is set 1`] = `
<section>
  <SectionHeading
    id="test"
    slotName="sectionToolbar"
    slotProps={
      Object {
        "classes": Object {},
        "name": "test",
        "slug": "test",
      }
    }
  >
    test
  </SectionHeading>
</section>
`;

exports[`should not render components list if not defined 1`] = `
<Styled(Section)
  isolated={false}
  name="No components"
  slug="no-components"
/>
`;

exports[`should not render sections if not defined 1`] = `
<Styled(Section)
  isolated={false}
  name="No sections"
  slug="no-sections"
/>
`;

exports[`should render component renderer 1`] = `
<Styled(Section)
  components={
    <Components
      components={Array []}
    />
  }
  content={
    <Examples
      examples={
        Array [
          Object {
            "content": "<button>OK</button>",
            "evalInContext": [Function],
            "type": "code",
          },
          Object {
            "content": "Hello *world*!",
            "type": "markdown",
          },
        ]
      }
      name="Foo"
    />
  }
  isolated={false}
  name="Foo"
  sections={
    <Sections
      sections={Array []}
    />
  }
  slug="foo"
/>
`;

exports[`should render component renderer with primary title 1`] = `
<Styled(Section)
  components={
    <Components
      components={Array []}
    />
  }
  content={
    <Examples
      examples={
        Array [
          Object {
            "content": "<button>OK</button>",
            "evalInContext": [Function],
            "type": "code",
          },
          Object {
            "content": "Hello *world*!",
            "type": "markdown",
          },
        ]
      }
      name="Foo"
    />
  }
  isolated={false}
  name="Foo"
  primary={true}
  sections={
    <Sections
      sections={Array []}
    />
  }
  slug="foo"
/>
`;

exports[`should render components list 1`] = `
<Styled(Section)
  components={
    <Components
      components={Array []}
    />
  }
  isolated={false}
  name="Components"
  slug="components"
/>
`;

exports[`should render sections if defined 1`] = `
<Styled(Section)
  isolated={false}
  name="Nested sections"
  sections={
    <Sections
      sections={Array []}
    />
  }
  slug="nested-sections"
/>
`;
